types/highcharts/index.d.ts

```diff
@@ -5910,7 +5910,7 @@ declare namespace Highcharts {
          *            [3, 7, 8]
          *
          */
-        data?: Array<number | [number, number] | [string, number] | [string, number, number] | [number, number, number] | DataPoint>;
+        data?: Array<number | null | [number, number] | [number, null] | [string, number] | [string, number, number] | [number, number, number] | DataPoint | null>;
```

Comment:
It doesn't say anywhere in the documentation that data can be null. Please revert.

Reply:
From the documentation:

> An array of numerical values. In this case, the numerical values will be interpreted as `y` options

> An array of arrays with 2 values. In this case, the values correspond to `x,y`.

In addition the type annotation at https://github.com/highcharts/highcharts/blob/8f802d8c7/js/parts/Series.js#L5613 seems to allow it

> `* @type      {Array<number|Array<(number|string),(number|null)>|null|*>}`

Reply:
Examples:

Data with number and null: https://jsfiddle.net/6vptua08/
Data with [number,number] and [number,null] pairs: https://jsfiddle.net/gzya3e7c/
Data with [string, number] and [string, null] (x value is inferred, in this case 1 per point): https://jsfiddle.net/gzya3e7c/2/